### ISSUE TITLE
Binding issue on pilosa

### ISSUE THREAD

**Issue by akshayagarwal2894** [pilosa/pilosa#710]
Pilosa config

bind = "IP:10101"
data-dir = "/opt/pilosa/data"
log-path = "/var/log/pilosa.log"
[cluster]
poll-interval = "2m0s"
replicas = 2
type = "http"
hosts = ["xyz1:10101" ,  “xyz2:10101” , ”xyz3:10101"]
internal-port = 11101
internal-hosts = [“xyz1”, “xyz2”, “xyz3”]

Pilosa is binded on all hosts when started despite specifying bind to some other ip in config.

Logs:

Pilosa v0.4.0
Using data from: /opt/pilosa/data
Listening as http://IP:10101

Logs show pilosa is listening on ip:port but when netstat is done it gives

tcp6       0      0 :::10101                :::*                    LISTEN      16725/pilosa
tcp6       0      0 :::11101                :::*                    LISTEN      16725/pilosa

**Comment by akshayagarwal2894** [comment]
There is also problem occuring in internal communication and replication of data accross nodes.
In logs it says sync beginning and sync complete but on checking other nodes data is not replicated.Can you please look into this issue as well.

**Comment by raskle** [comment]
This looks like an issue with the cluster configuration.  
internal-hosts should also specify the port per host.
i.e. 
internal-hosts = ["xyz1:11101", "xyz2:11101", "xyz3:11101"]
The bind string should exactly match one of the host names from "hosts" 
i.e. 
on node1
bind = "xyz1:10101"

I am starting a new ticket #712 to improve the error reporting that would catch this on startup.

**Comment by raskle** [comment]
We are working on #712  to improve the error reporting and documentation. @akshayagarwal2894 if the internal-hosts port specification change I described above fixes your issue, please go ahead and close this ticket - if not, let us know and we'll dig in further.

**Comment by akshayagarwal2894** [comment]
I was facing the same issue even after specifying the port for internal-hosts.The binding was still done on all interfaces.

**Comment by raskle** [comment]
@akshayagarwal2894 I wanted to see if you were still having any issue with your cluster setup after #750.  I believe this should take care of any remaining setup problems.  Let us know if you have any other questions, I'm going to close this issue.
